modules/mobile/app/controllers/mobile/v0/facilities_info_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require_dependency 'mobile/application_controller'
+
+module Mobile
+  module V0
+    class FacilitiesInfoController < ApplicationController
+      def index
+        facility_ids = @current_user.va_treatment_facility_ids + @current_user.cerner_facility_ids
+        facilities = appointments_proxy.fetch_facilities_from_ids(facility_ids.uniq, true)
+        adapted_facilities = facilities.map do |facility|
+          Mobile::V0::Adapters::FacilityInfo.new.parse(facility, @current_user, params)
+        end
+        sorted_facilities = sort(adapted_facilities, params[:sort])
+        render json: Mobile::V0::FacilitiesInfoSerializer.new(@current_user.id, sorted_facilities)
+      end
+
+      private
+
+      def appointments_proxy
+        Mobile::V0::Appointments::Proxy.new(@current_user)
+      end
+
+      def sort(facilities, sort_method)
+        case sort_method
+        when 'home', 'current'
+          facilities.sort_by(&:miles)
+        when 'alphabetical'
+          facilities.sort_by(&:name)
+        when 'appointments'
+          sort_by_recent_appointment(facilities)
+        else
+          raise Common::Exceptions::ValidationErrorsBadRequest.new(
+            detail: 'Invalid sort method', source: self.class.to_s
+          )
+        end
+      end
+
+      def sort_by_recent_appointment(facilities)
+        facilities.sort_by(&:name) # entries not in recent appointments will be sorted alphabetically
+        appointments = Mobile::V0::Appointment.get_cached(@current_user)
```

Comment:
Does getting cached appointments already come pre-sorted by `start_date_utc`?

Reply:
It definitely does for new appointments fetch. The legacy fetch doesn't have an explicit sort, but my assumption is yes since that's how they are displayed in the app. I could add a sort by to be 100% sure